src/wp-includes/blocks/index.php

```diff
@@ -40,26 +39,62 @@ function register_core_block_style_handles() {
 		'editorStyle' => 'editor',
 	);
 
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require $includes_path . 'blocks/blocks-json.php';
+	}
+
+	$files          = false;
+	$transient_name = 'wp_core_block_css_files';
 	/*
 	 * Ignore transient cache when the development mode is set to 'core'. Why? To avoid interfering with
 	 * the core developer's workflow.
 	 */
 	if ( ! wp_is_development_mode( 'core' ) ) {
-		$transient_name = 'wp_core_block_css_files';
-		$files          = get_transient( $transient_name );
-		if ( ! $files ) {
-			$files = glob( wp_normalize_path( __DIR__ . '/**/**.css' ) );
-			set_transient( $transient_name, $files );
+		$cached_files = get_transient( $transient_name );
+
+		/*
+		 * Check the validity of cached values.
+		 *  - Should match the current WordPress version
+		 *  - Should use relative paths for the files
+		 */
+		if ( is_array( $cached_files ) ) {
+			if ( ! isset( $cached_files['version'] ) || $cached_files['version'] !== $wp_version ) {
+				$files = false;
+			} elseif ( ! isset( $cached_files['files'] ) || empty( $cached_files['files'] ) || ! str_starts_with( reset( $cached_files['files'] ), 'blocks/' ) ) {
+				$files = false;
+			} else {
+				$files = $cached_files['files'];
+			}
```

Comment:
Related to https://github.com/WordPress/wordpress-develop/pull/5052/files#r1316063618, we can actually simplify this logic quite a bit more: `$files` is already `false` above, so no need to set it to `false` here in any circumstances.
```suggestion
		if (
			is_array( $cached_files )
			&& isset( $cached_files['version'] )
			&& $cached_files['version'] === $wp_version
			&& isset( $cached_files['files'] )
		) {
			$files = $cached_files['files'];
```